carta/session.py

```diff
@@ -391,6 +391,31 @@ def append_image(self, path, hdu="", complex=False, expression=ArithmeticExpress
             Arithmetic expression to use if appending a complex-valued image. The default is :obj:`carta.constants.ArithmeticExpression.AMPLITUDE`.
         """
         return Image.new(self, path, hdu, True, complex, expression)
+    
+    def load_stokes_hypercube(self, output_direcory, directories, files, polarization_type, hdu="", append=False):
+        """Open or append a new Stokes hypercube with the selected Stokes images.
+
+        Parameters
+        ----------
+        output_direcory : {0}
+            The direcory to output the hypercube, either relative to the session's current directory or an absolute path relative to the CARTA backend's root directory.
+        directories : {1}
+            The list of paths to the images, either relative to the session's current directory or an absolute path relative to the CARTA backend's root directory.
+        files : {2}
+            The list of images for making the hypercube.
+        polarization_type : {3}
+            The list of the polarization types assigned to the respective images.
+        hdu : {4}
+            The HDU to select inside the file.
+        append : {5}
+            Whether the hypercube should be appended. Default is ``False``.
+        """
+        stokes_files = []
+        for i in range(len(directories)):
```

Comment:
Here is a suggested implementation (partial and untested, and depends on the validation PR). Comments are for explanation purposes only:

```python
###### In constants.py, before Polarization (dict keys = strings matching Polarization enum names; values = ints matching the *protobuf* enum numbering):

PROTO_POLARIZATION = {
    "I":  1,
    "Q": 2,
    # (...)
    "PANGLE": 17,
}

# Then we add an init to the Polarization enum to add this index as an additional property (like in PaletteColor):

    def __init__(self, value):
        self.proto_index = PROTO_POLARIZATION[self.name]

###### In a new structs.py:

# (imports here...)

class StokesImage:

    # It should be fine to use this decorator here
    # We validate these fields when this object is constructed, so that later we only have to validate the object type
    @validate(Constant(Polarization), String(), String())
    def __init__(self, stokes, path, hdu=""):
        self.stokes = stokes
        # Here we only split the path for now; we will make it absolute later
        self.directory, self.file_name = posixpath.split(path)
        self.hdu = hdu
        
    def json(self):
        # This is returning the object which will actually be serialized by the serializer (in this case a dictionary).
        # Note the lookup of the protobuf index through the additional property on the enum:
        return {"directory": self.directory, "file": self.file_name, "hdu": self.hdu, "polarizationType": self.stokes.proto_index}
        
###### In session.py, in your new function

    # Now we can have one list as the main parameter
    # Does it make sense to require at least two images?
    @validate(IterableOf(InstanceOf(StokesImage), min_size=2), Boolean())
    def load_stokes_hypercube(self, stokes_images, append=False):
        # Now that we're in the session, we can normalize the paths:
        for image in stokes_images:
            image.directory = self.resolve_file_path(image.directory)
        # If it really doesn't matter what the output directory is, we could use a macro
        # to get the starting directory of the file browser
        # which is used in several frontend actions that use addFrame to open generated images
        output_directory = Macro("fileBrowserStore", "startingDirectory")
        # If I'm reading the frontend code correctly, the single hdu parameter is the output hdu, and we can just use the default
        output_hdu = ""

        # Now you can get the command as before
        command = "appendConcatFile" if append else "openConcatFile"
        self.call_action(command, stokes_images, output_directory, output_hdu)
```

I'm assuming that you need the protobuf polarization enum numbering in the objects that you're passing to the action (this should be tested with the polarizations with indices that differ (xx, xy, etc.)).

So the API for the user would look something like this:

```python
session.load_stokes_hypercube([
    StokesImage(Polarization.I, "path/to/imageI.fits"),
    StokesImage(Polarization.Q, "path/to/imageQ.fits"),
    StokesImage(Polarization.U, "path/to/imageU.fits"),
    StokesImage(Polarization.V, "path/to/imageV.fits"),
])
```

All the optional parameters are hidden, there's one parameter for each image path, and the same polarization enums are used here as in the existing functions (the mapping is done internally).

If we like this approach, we can perhaps do something similar for the function which opens multiple images (but with a different wrapper object).